src/Configuration/Dotenv/ParsedVariable.hs

```diff
@@ -0,0 +1,55 @@
+module Configuration.Dotenv.ParsedVariable (ParsedVariable(..),
+                                            VName,
+                                            VValue(..),
+                                            VContents,
+                                            VFragment(..),
+                                            interpolateParsedVariables) where
+
+import Control.Monad (foldM)
+#if MIN_VERSION_optparse_applicative(0,13,0)
+import Data.Functor ((<$>))
+#endif
+import Control.Applicative ((<|>))
+import System.Environment (lookupEnv)
+
+data ParsedVariable
+  = ParsedVariable VName VValue deriving (Show, Eq)
+
+type VName = String
+
+data VValue
+  = Unquoted VContents
+  | SingleQuoted VContents
+  | DoubleQuoted VContents deriving (Show, Eq)
+
+type VContents = [VFragment]
+
+data VFragment
+  = VInterpolation String
+  | VLiteral String deriving (Show, Eq)
+
+interpolateParsedVariables :: [ParsedVariable] -> IO [(String, String)]
```

Comment:
I'll just type up the response as best as I can remember it.

As I said, I'm quite new to Haskell, so I'm not exactly well-versed in keeping functions as pure as possible. I took a quick look again to reacquaint myself with the code I wrote here 2 months ago, and the IO does seem to be necessary in all the functions (and the necessity derives from the fact that we access the environment).

I'd love some advice on maybe what abstraction would work well to achieve the effect I'm trying to achieve here, but `foldM` was the simplest way I could see to write the `interpolateParsedVariables` function.

I can't really be bothered to explain why the rest of them need IO as well one by one all over again, but if you have any ideas, I'm game to hear them.